code-challenges.test.js

```diff
@@ -61,7 +84,45 @@ var hand3 = [5, 5, 5, 5, 4]
 // Expected output: false
 
 
+describe ("when 'fullHouse' is called", () => {
+    let hand1 = [5, 5, 5, 3, 3]
+    let result1 = true
+    let hand2 = [5, 5, 3, 3, 4]
+    let result2 = false
+    let hand3 = [5, 5, 5, 5, 4]
+    let result3 = false
+    it ("decides full house", () => {
+        expect (fullHouse(hand1)).toEqual(result1)
+    })
+    it ("replaces full house", () => {
+        expect (fullHouse(hand2)).toEqual(result2)
+    })
+    it ("replaces full house", () => {
+        expect (fullHouse(hand3)).toEqual(result3)
+    })
+})
+
 
 
 
 // b) Create the function that makes the test pass.
+
+
+const fullHouse = (hand) => {
+    for(let i=0; i<hand.length; i++){
```

Comment:
This is definitely one way to check for some of the test cases, but this logic is dependent on values within the array having a certain order. For instance, an array like [2,3,2,3,3] would return false but it is a full house

Reply:
also it seems that the for-loop isn't necessary because the logic within the for-loop isn't dependent on the iteration or accessing the array using an index